lib/kwtool_web/router.ex

```diff
@@ -7,23 +7,40 @@ defmodule KwtoolWeb.Router do
     plug :fetch_flash
     plug :protect_from_forgery
     plug :put_secure_browser_headers
+    plug KwtoolWeb.Plugs.SetCurrentUser
+  end
+
+  pipeline :auth do
+    plug Kwtool.Accounts.Pipeline
+  end
+
+  pipeline :ensure_auth do
+    plug Guardian.Plug.EnsureAuthenticated
```

Comment:
I think when using guardian we can

1. Move the `KwtoolWeb.Plugs.SetCurrentUser` to after `EnsureAuthenticated`
2. In `SetCurrentUser`, use guardian method to get the current user, probably similar to the code that we get the current user before

```elixir
 Kwtool.Accounts.Guardian.Plug.current_resource(conn)
```

This is to use the guardian to retrieve the resource. Because the strategy to get the resource is already defined in ` Kwtool.Accounts.Guardian` 👍

Reply:
yes, I see the problem:
I am using `Guardian` however trying to `get`/`put` the user to the session manually so it looks like not the correct way to do it.

I just refactor for the get current user and authentication using `Guardian` only. And removed the custom Plug `SetCurrentUser`.

Thanks for clarification, I was confused that point before.

Reply:
Oh ok, Actually I am thinking of keeping the current user plug but in that plug we need to use the guardian method to get current user


The plug:
```elixir
pipeline :ensure_auth do
    plug Guardian.Plug.EnsureAuthenticated
    plug KwtoolWeb.Plugs.SetCurrentUser
end
```

This is because the guardian plug will load the resource using the strategy we defined. Then current user plug we can fetch the user by guardian method. Maybe something like this

```elixir
def call(conn, _params) do
    current_user = Kwtool.Accounts.Guardian.Plug.current_resource(conn)

    if (current_user) do
      conn
      |> assign(:current_user, current_user)
      |> assign(:user_signed_in?, true)
    else
      conn
      |> assign(:current_user, nil)
      |> assign(:user_signed_in?, false)
    end
  end
```

And other places, we can just call `conn.assigns.current_user` What do you think?

Reply:
hi @rosle !

yup, I totally agree with that, :)

I just reverted the Plug, however, I put it to the `auth` to make it work for logged in also not logged in user 

```
  pipeline :auth do
    plug Kwtool.Accounts.Pipeline
    plug KwtoolWeb.Plugs.SetCurrentUser
  end
```

Because I need to check `user_signed_in?` in the main layout also.